### PR TITLE
Add chapter on neural network backpropagation.

### PR DESCRIPTION
**Pull request by jayanthkoushik**
Hey! I've been following this course online and found the 'real-valued circuit' approach to gradient computation really useful. However, the notes don't really go over using this intuitive method to derive the gradient for a full neural network.

I spent some time working out the backpropagation equations by considering the network as a circuit. The process really helped me understand the whole concept really well, so I figured I'd put together my notes and make the derivation formal.

The notes are in a file by themselves: backprop-derivation.md, and there's a live version at http://jayanthkoushik.github.io/cs231n.github.io/backprop-derivation/. I'm not sure  if you guys accept such contributions from people outside Stanford, but I though I'd share anyway in case it can be useful for someone else too.

Cheers!

### CODE REVIEW

**Comment by karpathy** [comment]
Thanks! I'll be going through backprop notes in ~week or so I'll take a more closer look then. -Andrej
